go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -321,6 +394,23 @@ func (h *networkClusterControllerEventHandler) DeleteResource(obj, cachedObj int
 			return fmt.Errorf("could not cast obj of type %T to *knet.Node", obj)
 		}
 		return h.ncc.nodeAllocator.HandleDeleteNode(node)
+	case factory.IPAMClaimsType:
+		ipamClaim, ok := obj.(*ipamclaimsapi.IPAMClaim)
+		if !ok {
+			return fmt.Errorf("could not cast obj of type %T to *ipamclaimsapi.IPAMClaim", obj)
+		}
+
+		ipAllocator := h.ncc.subnetAllocator.ForSubnet(h.ncc.NetInfo.GetNetworkName())
+		err := h.ncc.ipamClaimReconciler.Reconcile(ipamClaim, nil, ipAllocator)
+		if errors.Is(err, persistentips.ErrIgnoredIPAMClaim) {
+			klog.V(5).Infof(
```

Comment:
Would it be more useful to have a log when the actual release happens?

Reply:
Done. Is this OK now ?